### PR TITLE
Colorize status bar item hover

### LINKED ISSUE #60: Colorize status bar item hover background
The current colorization of the status bar includes the background and foreground. The status bar will contain several items that have hover states and most themes will set the `statusBarItem.hoverBackground` to something relatively close to the `statusBar.background` to display the interactivity of the hover state, but minimize the visual change that occurs. Currently Peacock does not set this color when the user has chosen to affect the status bar and it can result in some intense color changes on hover.

As an example, in my current Atom One Dark theme the status bar item hover background is a darker gray #2c313a. When using this in with the Angular Red coloring the difference is stark

screen shot 2019-03-06 at 9 43 24 am

## Proposal

Leverage the tinycolor library to make either a slightly lighter or slightly darker hover state based on the color of the status bar background. When the user chooses to affect the status bar, the `statusBarItem.hoverBackground` would also be set to this color. With the same conditions as above that could look like:

screen shot 2019-03-06 at 2 11 32 pm

## Version Information

macOS 10.14
VS Code 1.31 stable
Peacock 0.7.0

### PR DESCRIPTION
**Pull request by musicfuel**
## Purpose

- Colorize the Status Bar Item hover background as part of the process that Peacock manages when that element is being affected to improve visual coherence of the color that Peacock applies when the user interacts with the element.

## What

- Closes #60 
- The existing behavior of Peacock in 0.7.0 and below would not colorize the Status Bar item hover color, so when the user interacted with the status bar the hover color would revert to the theme hover color which could be a very different color from the one Peacock applied.
- When the Status Bar is affected and colorized, the item hover background color will be set to a slightly different shade of the background color to provide the visual cue of the hover state, but not make the difference a potentially jarring one.
  - If the background is a light color, the hover will be slightly darker.
  - If the background is a dark color, the hover will be slightly lighter.

## How to Test

- Tests added for status bar hover color

## Checklist of changes included

- [x] CHANGELOG updated
- [ ] README updated
- [x] Tests updated/added
- [x] Prettier formatting
- [x] console.log removed
- [x] Dead code removed
- [x] Unnecessary comments removed
- [ ] Images updated (if applicable)

### CODE REVIEW

**Comment by johnpapa** [comment]
Found 4 tests are failing. Might be due to merge - can you take a look?


```
         set adjustment to lighten for an affected element is lighter color:
      AssertionError [ERR_ASSERTION]: Expected original brightness 86.707 to be less than 0, but was greater
      + expected - actual
      -false
      +true
      
      at Object.<anonymous> (/Users/papa/_git/vscode-peacock/src/test/extension.test.ts:727:14)
      at Generator.next (<anonymous>)
      at fulfilled (/Users/papa/_git/vscode-peacock/out/test/extension.test.js:8:58)
  4) Extension Basic Tests
base.js:240
       Element adjustments
         can adjust the color of an affected elements independently:
     AssertionError [ERR_ASSERTION]: '#d04649' == undefined
      at Object.<anonymous> (/Users/papa/_git/vscode-peacock/src/test/extension.test.ts:750:14)
      at Generator.next (<anonymous>)
      at fulfilled (/Users/papa/_git/vscode-peacock/out/test/extension.test.js:8:58)
```

File: src/test/extension.test.ts [hunk 1]
```diff
@@ -565,8 +611,6 @@ suite('Extension Basic Tests', () => {
         let config = getPeacockWorkspaceConfig();
         assert.ok(!config[ColorSettings.activityBar_badgeBackground]);
         assert.ok(!config[ColorSettings.activityBar_badgeBackground]);
-
-        await updateAffectedElements(allAffectedElements);
```
Comment (musicfuel) [inline src/test/extension.test.ts]: This probably still should be here to reset the state before leaving the test

Comment (johnpapa) [reply src/test/extension.test.ts]: thanks. that worked.

probably should be in setup or teardown moving forward

Comment (musicfuel) [reply src/test/extension.test.ts]: So to do specific setup for a single test should it be wrapped in a `suite()` where `suiteSetup()` and `suiteTeardown()` are available?

Comment (johnpapa) [reply src/test/extension.test.ts]: Thinking out loud:

1. suite would mean doing it once before and after the suite

2. test level is before and after each test

3. if it only affect one test, inside the test is ok.

I occasionally review the tests to see where these are. I think a refactor of the rest may b in order soon. There are a ton of them in one massive suite. I may want to create a few suites that call reusable functions to do setup/teardown

**Pull request merged**